src/lib.rs

```diff
@@ -75,6 +75,20 @@ pub struct ConnectParams {
     pub p2p_socket: Option<SocketAddrV4>,
 }
 
+impl ConnectParams {
+    /// Parses the cookie file content
+    fn parse_cookie(content: String) -> anyhow::Result<(Option<String>, Option<String>)> {
+        let values: Vec<&str> = content.splitn(2, ':').collect();
+        Ok((Some(values[0].into()), Some(values[1].into())))
```

Comment:
This should have been `values.get(0)` otherwise there is little sense in returning an option

Reply:
Sorry about that 😅

Reply:
Eheh half my fault for not finding it during review